presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java

```diff
@@ -708,6 +717,144 @@ else if (writerTarget instanceof ExecutionWriterTarget.RefreshMaterializedViewHa
         }
     }
 
+    // Returns RDD for specified fragmented subplan
+    // This method ensures that RDD is created only once for a sub-plan, where identity is determined by fragment id
+    protected <T extends PrestoSparkTaskOutput> RddAndMore<T> createRddFragmented(SubPlan subPlan, TableWriteInfo tableWriteInfo)
```

Comment:
createRddFragmented isn't a very clear name to me. maybe call this `createRddForSubPlan`.